Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -768,17 +454,17 @@ class CombinatorialKalmanFilter {
       auto currentBranch = result.activeBranches.back();
       TrackIndexType prevTip = currentBranch.tipIndex();
 
-      // Create trackstates for all source links (will be filtered later)
       using TrackStatesResult =
           Acts::Result<CkfTypes::BranchVector<TrackIndexType>>;
       TrackStatesResult tsRes = TrackStatesResult::success({});
-      if (hasMeasurements) {
-        auto [slBegin, slEnd] = *slRange;
-
-        tsRes = trackStateCandidateCreator(
+      if (isSensitive) {
+        // extend trajectory with measurements associated to the current surface
+        // which may create extra trajectory branches if more than one
+        // measurement is selected.
+        tsRes = m_extensions.createTrackStates(
             state.geoContext, *calibrationContextPtr, *surface, boundState,
-            slBegin, slEnd, prevTip, *result.trackStates,
-            result.trackStateCandidates, *result.trackStates, logger());
+            prevTip, result.trackStateCandidates, *result.trackStates,
+            logger());
```

Comment:
Dear master coder abitai,
The  proposed debug message is useless since the cache is expected to be empty before calling createTrackStates. But the proposal for the error message makes indeed sense.

Reply:
Applied improvement to error message.